### PR TITLE
Add support for tags in the commitlog

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: persist/fs/commitlog/writer.go [hunk 1]
```diff
@@ -27,12 +27,15 @@ import (
 	"os"
 	"time"
 
+	"github.com/m3db/m3x/ident"
```
Comment (robskillington) [inline persist/fs/commitlog/writer.go]: nit: Import order grouping.

File: persist/fs/commitlog/reader.go [hunk 2]
```diff
@@ -405,12 +409,37 @@ func (r *reader) decodeAndHandleMetadata(
...
+	var (
+		tags        ident.Tags
+		tagBytesLen = len(decoded.Tags)
+	)
+	if tagBytesLen != 0 {
+		tagsBytes := r.checkedBytesPool.Get(len(decoded.Tags))
+		tagsBytes.IncRef()
+		tagsBytes.AppendAll(decoded.Tags)
+		tagsBytes.DecRef()
+		tagDecoder.Reset(tagsBytes)
```
Comment (robskillington) [inline persist/fs/commitlog/reader.go]: You're going to clone the bytes below, perhaps best to keep a `checked.Bytes` around and pass it to the `tagDecoder`?  Will avoid the pooling/locking of tag bytes.

e.g. 
```go
// some cached checked.Bytes somewhere
tagBytes := checked.NewBytes(nil, nil)


tagBytes.Reset(decoded.Tags)
tagDecoder.Reset(tagBytes)

// .. clone the tags from the decoder
```

File: storage/bootstrap/bootstrapper/commitlog/source.go [hunk 3]
```diff
@@ -348,7 +350,7 @@ func (s *commitLogSource) mergeShard(
 			if shardResult == nil {
 				shardResult = result.NewShardResult(len(unmergedShard.encodersBySeries), s.opts.ResultOptions())
 			}
-			shardResult.AddSeries(unmergedBlocks.id, nil, seriesBlocks) // FOLLOWUP(prateek): include tags in commit log reader
+			shardResult.AddSeries(unmergedBlocks.id, unmergedBlocks.tags, seriesBlocks)
```
Comment (robskillington) [inline storage/bootstrap/bootstrapper/commitlog/source.go]: Nice, perhaps add a similar bootstrapper integration test for the commit log similar to the fs one?

File: persist/fs/commitlog/reader.go (hunk 2 above)
Comment (richardartoul) [reply persist/fs/commitlog/reader.go]: oh duh, thanks

File: storage/bootstrap/bootstrapper/commitlog/source.go (hunk 3 above)
Comment (richardartoul) [reply storage/bootstrap/bootstrapper/commitlog/source.go]: Discussed in person, gonna hold off and add one for the index in the next P.R

File: persist/schema/types.go [hunk 4]
```diff
@@ -92,4 +92,5 @@ type LogMetadata struct {
 	ID        []byte
 	Namespace []byte
 	Shard     uint32
+	Tags      []byte
```
Comment (robskillington) [inline persist/schema/types.go]: nit: Should we call this `EncodedTags`?  I don't feel strongly, however it is what it's called in the `IndexEntry`.

**Pull request merged**